pydantic/_internal/_model_construction.py

```diff
@@ -307,4 +310,6 @@ def apply_alias_generator(config: ConfigDict, fields: dict[str, FieldInfo]) -> N
             if not isinstance(alias, str):
```

Comment:
we need to fix this, and update the signature of `f'alias_generator`

Reply:
we also need a way for `alias_generator` to set only one of the aliases.

Perhaps we should change the signature to:

```py
class AliasGeneratorReturn(TypedDict, total=False):
    validation_alias: str | list[str | int] | list[list[str | int]]
    serialization_alias: str
```

Then the return type should be `AliasGeneratorReturn | str`.

Reply:
Based on [@dmontagu comment](https://github.com/pydantic/pydantic/pull/5433#issuecomment-1505349710), `alias_generator` should still return `str`